tsdb/tsdbutil/histogram.go

```diff
@@ -53,7 +51,7 @@ func GenerateTestHistogram(i int) *histogram.Histogram {
 
 func GenerateTestGaugeHistograms(n int) (r []*histogram.Histogram) {
 	for x := 0; x < n; x++ {
-		r = append(r, GenerateTestGaugeHistogram(rand.Intn(n)))
+		r = append(r, GenerateTestGaugeHistogram(x))
```

Comment:
It changed the size of the generated histograms which causes histogram chunks to be split differently between test runs.

I see your point about the change making it effectively a counter though. What if we do `x % n` instead of `x` there?

Reply:
Never mind, just realized `x % n` is essentially the same thing...maybe `x % (n/2)`?

Reply:
Updated [here](https://github.com/prometheus/prometheus/pull/12054/commits/5fdb7372427d7a35e1fd3e86fdf547d059d53956); I ended up using a sine wave to simulate gauge-iness but also be consistent for the tests.